pattern/variable/Variable.java

```diff
@@ -74,6 +81,31 @@ public ThingVariable asThing() {
         throw GraknException.of(INVALID_CASTING, className(this.getClass()), className(ThingVariable.class));
     }
 
+    public void addHints(Set<Label> labels) {
+        typeHints.addAll(labels);
+    }
+
+    public void retainHints(Set<Label> labels) {
+        typeHints.retainAll(labels);
+    }
+
+    public void removeHint(Label label) {
+        assert !(isSatisfiable && typeHints.isEmpty());
+        typeHints.remove(label);
+    }
+
+    public Set<Label> typeHints() {
+        return typeHints;
+    }
+
+    public boolean isSatisfiable() {
+        return isSatisfiable;
+    }
+
+    public void setSatisfiability(boolean isSatisfiable) {
```

Comment:
i'd just call this `setSatisfiable` to only have 1 terminology

Reply:
Changed to `setIsSatisfiable`.